lib/appsignal/oban.ex

```diff
@@ -0,0 +1,172 @@
+defmodule Appsignal.Oban do
+  require Appsignal.Utils
+
+  @tracer Appsignal.Utils.compile_env(:appsignal, :appsignal_tracer, Appsignal.Tracer)
+  @span Appsignal.Utils.compile_env(:appsignal, :appsignal_span, Appsignal.Span)
+  @appsignal Appsignal.Utils.compile_env(:appsignal, :appsignal, Appsignal)
+
+  @moduledoc false
+
+  require Logger
+
+  def attach do
+    handlers = %{
+      [:oban, :job, :start] => &__MODULE__.oban_job_start/4,
+      [:oban, :job, :stop] => &__MODULE__.oban_job_stop/4,
+      [:oban, :job, :exception] => &__MODULE__.oban_job_exception/4,
+      [:oban, :engine, :insert_job, :start] => &__MODULE__.oban_insert_job_start/4,
+      [:oban, :engine, :insert_job, :stop] => &__MODULE__.oban_insert_job_stop/4,
+      [:oban, :engine, :insert_job, :exception] => &__MODULE__.oban_insert_job_exception/4
+    }
+
+    for {event, fun} <- handlers do
+      case :telemetry.attach({__MODULE__, event}, event, fun, :ok) do
+        :ok ->
+          _ = Appsignal.IntegrationLogger.debug("Appsignal.Oban attached to #{inspect(event)}")
+
+          :ok
+
+        {:error, _} = error ->
+          Logger.warn("Appsignal.Oban not attached to #{inspect(event)}: #{inspect(error)}")
+
+          error
+      end
+    end
+  end
+
+  def oban_job_start(
+        _event,
+        _measurements,
+        %{job: job},
+        _config
+      ) do
+    span = @tracer.create_span("oban")
+
+    span
+    |> @span.set_name(to_string(job.worker))
```

Comment:
I noticed that we do not include the function name where the error occurred in, only the worker name. Is it always `perform` and can we append it, or can it be dynamic and difficult to retrieve?

So that it would become: `AppName.WorkerName.perform`
(Or are we still using the `#` legacy symbol for action names?)

Reply:
We also add `#perform` now for other libraries. That way people know exactly where to look. Not a blocker for me, but I'd like us to think about consistency like that.